File: pir.py
import RPi.GPIO as GPIO # type: ignore

class PIR:
    def __init__(self,pin):
        self.pin = pin
        GPIO.setmode(GPIO.BCM)
        GPIO.setup(self.pin,GPIO.IN)
    
    def digitalRead(self) -> bool:
        return GPIO.input(self.pin)
    
    def cleanup(self):
        GPIO.cleanup(self.pin)

File: cpp_code/src/pir.cpp
#include "pir.h"
#include <wiringPi.h>

PIR::PIR(int p):pin(p){
    wiringPiSetupGpio();
    pinMode(pin, INPUT);
}

bool PIR::digitalRead() noexcept{
    return ::digitalRead(pin) != 0;
}

void PIR::cleanup() noexcept{
}
